src/core/server/http/http_service.ts

```diff
@@ -94,6 +98,23 @@ export class HttpService implements CoreService<HttpServiceSetup, HttpServiceSta
     };
   }
 
+  public async createServer(config: HttpConfig) {
```

Comment:
This won't be accessible to plugins. Only methods on the `HttpStart` and `HttpSetup` interfaces are accessible. You'll want to return this function in the value returned by `setup` probably.

Reply:
Should I break this up or are we ok with this nested array generation? 

`await Promise.all([...this.secondaryServers.values()].map(s => s.stop()));`